### PR TITLE
fix #251: verify signatures of SNS messages

### LINKED ISSUE #251: Use SES for emails
On local machines and the heroku dev instance, we are using mailgun for emails.

For staging & prod we will need to use SES. To use SES, we will need to:

1. [x] Change the SMTP config to SES (See [bug 1480435](https://bugzilla.mozilla.org/show_bug.cgi?id=1480435))
2. [x] Specify a configuration set with an SMTP header: `X-SES-CONFIGURATION-SET: <ConfigSet>` (See [docs](https://docs.aws.amazon.com/ses/latest/DeveloperGuide/using-configuration-sets-in-email.html))
3. [x] Add a `/ses/notification` endpoint to handle email bounce notifications and delete those users from our subscribers
4. [ ] Add authentication (HTTP basic auth or digital signature verification) to `/ses/notification`

### PR DESCRIPTION
**Pull request by groovecoder**

### CODE REVIEW

File: middleware.js [hunk 1]
```diff
@@ -1,25 +1,5 @@
 "use strict";
 
-const auth = require("basic-auth");
-
-const AppConstants = require("./app-constants");
-
-
-function checkSESAuth (user, pass) {
-  return user === AppConstants.SES_USERNAME && pass === AppConstants.SES_PASSWORD;
```
Comment (pdehaan) [inline middleware.js]: Are we using `SES_USERNAME` or `SES_PASSWORD` still?


```sh
$ git grep SES_USERNAME
  # .env-dist:SES_USERNAME=""
  # app-constants.js:  "SES_USERNAME",
  # middleware.js:  return user === AppConstants.SES_USERNAME && pass === AppConstants.SES_PASSWORD;

$ git grep SES_PASSWORD
  # .env-dist:SES_PASSWORD=""
  # app-constants.js:  "SES_PASSWORD",
  # middleware.js:  return user === AppConstants.SES_USERNAME && pass === AppConstants.SES_PASSWORD;
```

Comment (groovecoder) [reply middleware.js]: Good catch! We can remove them now.

File: package.json [hunk 2]
```diff
@@ -38,6 +39,7 @@
     "jest": "^23.5.0",
     "jest-tap-reporter": "^1.9.0",
     "node-mocks-http": "^1.7.0",
+    "nodemon": "^1.18.3",
```
Comment (pdehaan) [inline package.json]: Are we using **nodemon** somewhere?
I didn't see it in any npm scripts or used anywhere specifically.

Comment (groovecoder) [reply package.json]: I started using it locally. This PR includes a new `nodemonConfig` section in `package.json` so we can use `nodemon server.js` to automatically refresh the `server.js` process when files are changed. I'll add it to the README>

**Comment by groovecoder** [comment]
Added nodemon to README. Going to merge this before it bit-rots.

**Pull request merged**